iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/RegistryManagerTests.java

```diff
@@ -459,35 +459,42 @@ public void apply_configuration_e2e() throws Exception
     public void deviceCreationWithDeviceScope() throws IOException, InterruptedException, IotHubException, URISyntaxException
     {
         // Arrange
-        RegistryManagerTestInstance testInstance = new RegistryManagerTestInstance();
-        deleteDeviceIfItExistsAlready(testInstance.registryManager, testInstance.deviceId);
+        String edge1Id = deviceIdPrefix + UUID.randomUUID().toString();
```

Comment:
So one is edge device, one leaf?

Reply:
This test nests a leaf device and an edge device under another edge, as the data model is different for a device or edge that is a child.